src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -33,7 +33,7 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool up
     if (!i_target.isValid() || !i_target->IsInWorld())
         return;
 
-    if (owner->HasUnitState(UNIT_STATE_NOT_MOVE))
+    if (owner->HasUnitState(UNIT_STATE_NOT_MOVE | UNIT_STATE_CASTING))
```

Comment:
I've work on this too and there is no triggered mask value or spellflag that allows it on 3.3.5